slash/react/src/Form/Select/__tests__/Select.test.tsx

```diff
@@ -0,0 +1,97 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import { axe } from "jest-axe";
+import { Select } from "../Select";
+
+const options = [
+  { value: "fun", label: "For fun" },
+  { value: "work", label: "For work" },
+  { value: "drink", label: "For drink" },
+];
+
+describe("Select", () => {
+  it("should have option and default class", async () => {
+    // Act
+    render(<Select mode="default" options={options} />);
+
+    // Asser
+    const selectInput = await screen.findByRole("combobox");
+    await screen.findByRole("option", { name: /- select -/i });
+    await screen.findByRole("option", { name: /for fun/i });
+    await screen.findByRole("option", { name: /for work/i });
+    await screen.findByRole("option", { name: /for drink/i });
+
+    expect(selectInput).toHaveClass("af-form__input-select");
+  });
+
+  it("should have mode base", async () => {
+    // Act
+    render(<Select mode="base" options={options} />);
+
+    // Assert
+    await screen.findByRole("combobox");
+    const selectOption = screen.queryByRole("option", { name: /- select -/i });
+    await screen.findByRole("option", { name: /for fun/i });
+    await screen.findByRole("option", { name: /for work/i });
+    await screen.findByRole("option", { name: /for drink/i });
+
+    expect(selectOption).not.toBeInTheDocument();
+  });
+
+  it("should mode base to be required", async () => {
+    // Act
+    render(<Select mode="base" required options={options} />);
+
+    // Assert
+    const selectBaseInput = await screen.findByRole("combobox");
+    expect(selectBaseInput).toBeRequired();
+  });
+
+  it("should mode base to be required with classModifier", async () => {
+    // Act
+    render(<Select mode="base" classModifier="required" options={options} />);
+
+    // Assert
+    const selectBaseInput = await screen.findByRole("combobox");
+    expect(selectBaseInput).toBeRequired();
+  });
+
+  it("should mode default have default value", async () => {
+    // Act
+    render(<Select mode="default" defaultValue="fun" options={options} />);
+
+    // Assert
+    const selectBaseInput = await screen.findByRole("combobox");
+    expect(selectBaseInput).toHaveValue("fun");
+  });
+
+  it("should mode default have onChange attribut", async () => {
+    // Act
+    render(
+      <Select
+        mode="default"
+        onChange={() => console.log("Some change")}
+        defaultValue="fun"
+        options={options}
+      />,
+    );
+
+    // Assert
+    const selectBaseInput = await screen.findByRole("combobox");
+    fireEvent.change(selectBaseInput, { target: { value: "drink" } });
+  });
```

Comment:
Je ne sais pas s'il est nécessaire de tester une prop native, mais s'il y a une surcouche d'onChange tu peux plutôt faire ça pour la tester : 

```suggestion
  it("should mode default have onChange attribut", async () => {
    const onChangeFn = vi.fn();
    // Act
    render(
      <Select
        mode="default"
        onChange={onChangeFn}
        defaultValue="fun"
        options={options}
      />,
    );

    // Assert
    const selectBaseInput = screen.getByRole("combobox");
    userEvent.selectOptions(selectBaseInput, { target: { value: "drink" } });
    expect(onChangeFn).toHaveBeenCalledWith(); // ici tu pourras mettre avec quels paramètres onChangeFn aura été appelé
  });
```

Dans la suggestion au dessus j'ai remplacé `fireEvent` par `userEvent` car ça reproduit mieux le comportement utilisateur, ce qui sera plus fiable pour les tests.

Reply:
Merci pour l'astuce :)